### ISSUE TITLE
Keras with data directory in AWS S3?

### ISSUE THREAD

**Issue by parkerzf** [fchollet/keras#4913]
I want to try a Keras model in AWS EC2 instance with dataset in AWS S3. 

However, I cannot directly use flow_from_directory pointing to S3 url. Is there an alternative way to do it?

Thanks!

**Comment by marcj** [comment]
You need to download the files first. It would be way too slow to read directly from S3.

**Comment by parkerzf** [comment]
Thanks!

**Comment by jerheff** [comment]
We used TensorFlow's feature to form training batches from distributed files and fed these batches to Keras through the generator capability.

**Comment by parkerzf** [comment]
How do you do that? Could you provide a code sample?

**Comment by jerheff** [comment]
S3 latency is actually fairly low and if you have multiple processes fetching data it is not a problem (we've done this in other contexts). 

We had experimented with this over a year ago, so I misremembered slightly in that TensorFlow was not directly doing the S3 fetches.  We experimented with presenting S3 files as a file system (like https://fullstacknotes.com/mount-aws-s3-bucket-to-ubuntu-file-system/).   Our data size was small enough to fit on an EBS volume, though, so ultimately we went that route to remove a piece in the architecture.  Keep in mind that EBS reads are also going over the network just like S3 reads.
